raiden/network/rpc/smartcontract_proxy.py

```diff
@@ -17,6 +21,37 @@ class TransactionFailed(Exception):
         pass
 
 
+class ClienErrorInspectResult(Enum):
+    """Represents the action to follow after inspecting a client exception"""
+    PROPAGATE_ERROR = 1
+    INSUFFICIENT_FUNDS = 2
+
+
+def inspect_client_error(val_err: ValueError, eth_node: str) -> ClienErrorInspectResult:
+    # both clients return invalid json. They use single quotes while json needs double ones.
+    json_response = str(val_err).replace("'", '"')
+    try:
+        error = json.loads(json_response)
+    except json.JSONDecodeError:
+        return ClienErrorInspectResult.PROPAGATE_ERROR
+
+    insufficient_funds_geth = (
+        error['code'] == -32000 and
+        eth_node == EthClient.GETH and
+        'insufficient funds' in error['message']
+    )
+    insufficient_funds_parity = (
+        error['code'] == -32010 and
+        eth_node == EthClient.PARITY and
+        'Insufficient funds' in error['message']
+    )
+
+    if insufficient_funds_geth or insufficient_funds_parity:
```

Comment:
`if all(insufficient_funds_geth) or all(insufficient_funds_parity)`

Reply:
What purpose would this serve?

Reply:
Sorry, i explained it below. But in general the you were checking the existence of elements inside the tuples but not the actual values. My fix was to add `all` which should evaluate the items and return True if they're all true.

Reply:
:man_facepalming: nevermind, i just saw you were actually doing `(X and Y and Z)` while i thought you were constructing a tuple.